src/vs/workbench/parts/snippets/electron-browser/snippetsService.ts

```diff
@@ -203,9 +210,12 @@ class SnippetsService implements ISnippetsService {
 						continue;
 					}
 
-					if (this._files.has(validContribution.location.toString())) {
-						this._files.get(validContribution.location.toString()).defaultScopes.push(validContribution.language);
-
+					const resource = validContribution.location.toString();
+					if (this._files.has(resource)) {
+						const file = this._files.get(resource);
+						if (file.defaultScopes) {
```

Comment:
This was the one change I was unsure about. Not sure what we are supposed to do if we get here and `file.defaultScopes` is undefined. Don't do anything? Create a new array?

Reply:
Yeah, a new array should be created. On the other hand this is a rare corner case and only happens when different snippet contributions point to the same file.. We can also nuke that if and always let the last one win...